crates/chia-consensus/src/spendbundle_validation.rs

```diff
@@ -46,32 +46,54 @@ pub fn validate_clvm_and_signature(
             .flat_map(move |(condition, items)| {
                 let spend_clone = spend.clone();
                 items.iter().map(move |(pk, msg)| {
-                    (
-                        pk,
-                        make_aggsig_final_message(
-                            condition,
-                            msg.as_slice(),
-                            &spend_clone,
-                            constants,
-                        ),
-                    )
+                    let mut aug_msg = pk.to_bytes().to_vec();
+                    let msg = make_aggsig_final_message(
+                        condition,
+                        msg.as_slice(),
+                        &spend_clone,
+                        constants,
+                    );
+                    aug_msg.extend_from_slice(msg.as_ref());
+                    let aug_hash = hash_to_g2(&aug_msg);
+                    let pairing = aug_hash.pair(pk);
+
+                    let mut hasher = Sha256::new();
+                    hasher.update(pk.to_bytes());
+                    hasher.update(msg.as_slice());
+                    let hash: [u8; 32] = hasher.finalize();
```

Comment:
it would probably make sense to factor this out as a separate function, as this code is repeated.